Complete task

4Empty4 opened the pull request:
Add Ball Lottery classes
Add Colors enum class
Create Ball.toString Lottery.getRandomBall
Update ColorSupplier.getRandomColor
Run tests

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/ColorSupplier.java:
```diff
@@ -1,7 +1,18 @@
+
     public String getRandomColor() {
-        return null;
+        return randomColor.get(random.nextInt(size)).toString();
```
This violates checklist item #10: 'Use name() for getting String representation of enum constants'. It's better to use the `name()` method for enums, as `toString()` can be overridden and might not return the constant's name as expected.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Application.java:
```diff
@@ -2,6 +2,14 @@
     public static void main(String[] args) {
+        Lottery lottery = new Lottery();
+        Lottery lottery1 = new Lottery();
+        Lottery lottery2 = new Lottery();
```
This violates checklist item #4: "Don't create several instances of the same class if you can use only one instance for your purposes". A single `Lottery` instance is sufficient to create all three balls.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Application.java:
```diff
@@ -2,6 +2,14 @@
+
+        lottery.getRandomBall();
+        lottery1.getRandomBall();
+        lottery2.getRandomBall();
```
This violates checklist item #8: "Use for loop for creating several objects of the same class". Instead of calling the method three times manually, a `for` loop should be used to generate the three balls. This makes the code cleaner and easier to modify if you need to create a different number of balls.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Colors.java:
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public enum Colors {
+    red,blue,yellow,green,black,white
```
This violates checklist item #6: 'Don't forget how to name constants according to style guide (the same with naming of enum values)'. Enum constants should be written in `UPPER_SNAKE_CASE`, for example, `RED`, `BLUE`, `YELLOW`, etc.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Lottery.java:
```diff
@@ -0,0 +1,17 @@
+
+public class Lottery {
+    private final Random random = new Random();
+    private final Ball ball = new Ball();
```
A single `Ball` instance is being reused for every call to `getRandomBall()`. According to the requirements, a new ball should be created inside the `getRandomBall()` method each time it is invoked.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Lottery.java:
```diff
@@ -0,0 +1,17 @@
+    private final Ball ball = new Ball();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public String getRandomBall() {
```
The task requires this method to return a `Ball` object, but it currently returns a `String`. The method signature should be updated to `public Ball getRandomBall()`.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Lottery.java:
```diff
@@ -0,0 +1,17 @@
+
+    public String getRandomBall() {
+        ball.setColor(colorSupplier.getRandomColor());
+        ball.setNumber(random.nextInt(100));
```
This violates checklist item #9: 'All magic numbers in your code should be constants.' The number `100` should be declared as a named constant to improve readability and maintainability.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Lottery.java:
```diff
@@ -0,0 +1,17 @@
+    public String getRandomBall() {
+        ball.setColor(colorSupplier.getRandomColor());
+        ball.setNumber(random.nextInt(100));
+        System.out.println(ball.toString());
```
The task description specifies that the `main()` method is responsible for printing information about the balls. This method should only create and return a ball, without producing any console output.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Lottery.java:
```diff
@@ -0,0 +1,17 @@
+        ball.setColor(colorSupplier.getRandomColor());
+        ball.setNumber(random.nextInt(100));
+        System.out.println(ball.toString());
+        return ball.toString();
```
This line should return the newly created `Ball` object, not its string representation.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Your solution is on the right track, and the `Ball` class is well-implemented. However, several critical changes are needed before this can be approved, primarily in the `Lottery` and `Application` classes.

Here are the key issues to address:

1.  **`Lottery.getRandomBall()` Method:**
    *   The method must return a `Ball` object, but it currently returns a `String`.
    *   A **new** `Ball` instance must be created inside this method for each call. Currently, you are reusing a single instance.
    *   Printing to the console should be done in the `main` method, not here.

2.  **`Application.main()` Method:**
    *   You should create only one `Lottery` instance and use it to get all three balls. [CHECKLIST ITEM #4]
    *   Please use a `for` loop to generate the three balls instead of calling the method three times manually. [CHECKLIST ITEM #8]

Additionally, please review the checklist items regarding naming conventions for enum constants (`UPPER_SNAKE_CASE`) [CHECKLIST ITEM #6] and avoiding magic numbers [CHECKLIST ITEM #9].

Keep up the effort! Addressing these points will significantly improve your code.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Application.java:
```diff
@@ -2,6 +2,12 @@
     public static void main(String[] args) {
+        Lottery lottery = new Lottery();
+
+        for (int i = 0; i < 3; i++) {
```
This violates checklist item #9: 'All magic numbers in your code should be constants.' The number `3` here is a magic number. It's better to declare it as a named constant to improve readability and make it easier to change in the future. For example:

```java
private static final int BALLS_COUNT = 3;
// ...
for (int i = 0; i < BALLS_COUNT; i++) {
```

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Lottery.java:
```diff
@@ -0,0 +1,17 @@
+    public Ball getRandomBall() {
+        Ball ball = new Ball();
+        ball.setColor(colorSupplier.getRandomColor());
+        ball.setNumber(random.nextInt(100) + 1);
```
This violates checklist item #9: 'All magic numbers in your code should be constants.' The number `100` should be declared as a named constant to improve code clarity and maintainability. For example: `private static final int MAX_BALL_NUMBER = 100;`

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Excellent work on this iteration! You've successfully addressed almost all the previous feedback, and the core logic is now correct. There's just one final point to address before this can be approved: the use of 'magic numbers'.

This violates a core requirement [CHECKLIST ITEM #9]. Please fix the following:
- In `Application.java`, the number `3` in the `for` loop should be a named constant.
- In `Lottery.java`, the number `100` for generating the random number should also be a constant.

Declaring these values as constants, for example `private static final int MAX_BALL_NUMBER = 100;`, makes the code much easier to read and maintain. Once you've made these changes, your solution will be ready for approval!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Excellent work! You've successfully addressed the feedback from the previous review by replacing the magic numbers `3` and `100` with well-named constants. Your code is now clean, readable, and fully compliant with all task requirements and best practices. I am approving your solution, great job!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

4Empty4 committed:
Complete task
Add Ball Lottery classes
Add Colors enum class
Create Ball.toString Lottery.getRandomBall
Update ColorSupplier.getRandomColor
Run tests

4Empty4 committed:
Small fix after Git tests
Change Ball class style

4Empty4 committed:
Fix after Git tests
Change Ball class
  Now create only one ball
Change Colors style(UPPER-CASE)
Change Application
  Now use loop for, for creating new balls
Change CollorSupplier
  Now return name by name not .toString

4Empty4 committed:
Hot Fix
Add global variable for Application loop and Lottery random pull
